libs/s25main/postSystem/PostBox.cpp

```diff
@@ -63,6 +63,12 @@ const PostMsg* PostBox::GetMsg(unsigned idx) const
     return messages[idx].get();
 }
 
+const PostMsg* PostBox::GetCurrentMsg() const
+{
+    unsigned idx = currMsgIdx >= 0 && currMsgIdx < (int) numMessages ? currMsgIdx : numMessages - 1;
```

Comment:
Is the possible underflow for `numMessages == 0`  intended?

Maybe be explicit and do:
```
if(curMsgIdx < 0 || numMessages == 0)
    return nullptr;
```

Afterwards you can use `static_cast<unsigned>(currMsgIdx)` safely instead of the (C-style) cast to int.

Reply:
It is intended, since `GetMsg` checks that the index is within bounds and returns `nullptr` if not. But you're right that it's probably best to be explicit about it. 👍

Reply:
I added check to make sure the index is within bounds. Checking that `currMsgIdx < 0` - as you suggest - does not make sense though, since `currMsgIdx == -1` whenever the notification window is closed. In that case, we get the the most recent notification (`numMessages - 1`).